src/network/network_gamelist.cpp

```diff
@@ -29,56 +29,36 @@ int _network_game_list_version = 0; ///< Current version of all items in the lis
  * @param connection_string the address of the to-be added item
  * @return a point to the newly added or already existing item
  */
-NetworkGameList *NetworkGameListAddItem(const std::string &connection_string)
+NetworkGame *NetworkGameListAddItem(const std::string &connection_string)
 {
-	NetworkGameList *item, *prev_item;
-
 	/* Parse the connection string to ensure the default port is there. */
 	const std::string resolved_connection_string = ServerAddress::Parse(connection_string, NETWORK_DEFAULT_PORT).connection_string;
 
-	prev_item = nullptr;
-	for (item = _network_game_list; item != nullptr; item = item->next) {
-		if (item->connection_string == resolved_connection_string) return item;
-		prev_item = item;
-	}
+	/* Check if it's already added. */
+	auto it = std::ranges::find_if(_network_game_list, [&resolved_connection_string](const auto &item) { return item->connection_string == resolved_connection_string; });
```

Comment:
```suggestion
	auto it = std::ranges::find(_network_game_list, resolved_connection_string, &NetworkGame::connection_string);
```
Won't a 'simple' `std::ranges::find` with projection work here?

Reply:
Huh, I assumed that wouldn't work because it's a `vector<unique_ptr<NetworkGame>>`, not a `vector<NetworkGame>`.